Update docs for telegram bot for new config flow (#39341)
home-assistant/home-assistant.github.io: source/_integrations/telegram_bot.markdown

## Introduction - Telegram bot platforms

Platforms are Telegram bot implementations for managing communications with Telegram for sending and receiving messages.
When setting up this integration, you should specify the platform which fits your environment and use case.

### Broadcast

Telegram implementation to support **sending messages only**. Your Home Assistant instance does not have to be exposed to the internet and there is no polling to receive messages or commands sent to the bot.

### Polling

Telegram chatbot polling implementation.
Your Home Assistant instance does not have to be exposed to the internet.

### Webhooks

Telegram chatbot webhooks implementation as described in the Telegram [documentation](https://core.telegram.org/bots/webhooks).
This implementation allows Telegram to push updates directly to your server and requires your Home Assistant instance to be exposed to the internet.

## Prerequisites

- Create your Telegram bot and [retrieve the API key](/integrations/telegram). The `api_key` will be used for adding the bot to Home Assistant during integration setup.

> - If you are going to use the `Webhooks` platform, you must configure either [HTTPS](/integrations/http#ssl_certificate) or [trusted proxies](/integrations/http#trusted_proxies) to allow Telegram to connect to your Home Assistant instance.

Reviewer: If you're using Home Assistant Cloud from NabuCasa and have an external URL configured, I think it may be enough to extend the trusted networks configuration.

https://community.home-assistant.io/t/set-up-telegram-using-nabu-casa/136604/7

You could also have an external URL configured provided by another service.

API key:
  description: The API token of your bot.
Proxy URL:
  description: Proxy URL if working behind one, optionally including username and password. (`socks5://username:password@proxy_ip:proxy_port`).
{% endconfiguration_basic %}

### Webhooks configuration

If you have selected the `Webhooks` Telegram bot type, the integration setup will continue with the webhooks configuration step.
{% configuration_basic %}
URL:
  description: Allow to overwrite the external URL from the Home Assistant [configuration](/integrations/homeassistant/#editing-the-general-settings-in-yaml) for different setups (`https://<public_url>:<port>`).
Trusted networks:
  description: Telegram server access ACL as list.
{% endconfiguration_basic %}

{% include integrations/option_flow.md %}

The integration can be configured to use a default parse mode for messages.

{% configuration_basic %}
Parse mode:
  description: Default parser for messages if not explicit in message data, either `markdown` (legacy), `markdownv2` or `html`. Refer to Telegram's [formatting options](https://core.telegram.org/bots/api#formatting-options) for more information.
{% endconfiguration_basic %}

> ## Whitelisting chat IDs via Subentries

Reviewer: ```suggestion
## Allowlisting chat IDs via Subentries
```

A Telegram chat ID is a unique numerical identifier for an individual user (positive) or a chat group (negative integer).

> You must whitelist the chat ID for the Telegram bot before it can send/receive messages for that chat.

Reviewer: ```suggestion
You must allowlist the chat ID for the Telegram bot before it can send/receive messages for that chat.
```

> To whitelist the chatID, [retrieve the chat ID](http://localhost:4000/integrations/telegram#methods-to-retrieve-a-chat_id) and create a subentry:

Reviewer: ```suggestion
To allowlist the chatID, [retrieve the chat ID](/integrations/telegram#methods-to-retrieve-a-chat_id) and create a subentry:
```

1. Go to **{% my integrations title="Settings > Devices & services" %}**.
2. Select the Telegram bot integration.

> 3. Add a subentry via {% my integrations title="**Settings** > **Devices & services**" %}, click {% icon "mdi:dots-vertical" %} and select **Add allowed chat ID**.

Reviewer: We've already said which page to go to, in step 1.

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `config_entry_id`          | yes      | The config entry representing the Telegram bot to send the message. Defaults to the first config entry.|

Reviewer: I think we should clarify that the config entry id is only required when using more than one config entry.

Saying that it defaults to the first config entry may be confusing since there's no default when there are are more than one config entry.
